### ISSUE TITLE
Run's `=<exit-status>` syntax is still documented

### ISSUE THREAD

**Issue by jpluscplusm** [bats-core/bats-core#578]
**Describe the bug**

The `run =<expected-exit-status>` convenience syntax (documented [here](https://bats-core.readthedocs.io/en/stable/writing-tests.html#run-test-other-commands)) is broken, and does not work.

I believe anyone relying on the currently-documented syntax to assert a `run` command's exit status has been inadvertently testing bats' breakage, not their underlying command's success or failure.  

**To Reproduce**

Running `bats` with the attached test file below should produce 4 test failures, according to the docs. Instead, it produces 2 passes and 2 failures. The correctly-failing-failures demonstrate that the alternative `run -<expected-exit-status>` syntax *does* work as expected.

```
$ bats foo.bats 
foo.bats
 ✓ This test should fail, but will succeed, #1
   ## Command that 'run' ran was: '=0 false'
 ✓ This test should fail, but will succeed, #2
   ## Command that 'run' ran was: '=1 true'
 ✗ This test should fail, and will fail, #1
   (in test file foo.bats, line 25)
     `run -0 false' failed, expected exit code 0, got 1
 ✗ This test should fail, and will fail, #2
   (in test file foo.bats, line 29)
     `run -1 true' failed, expected exit code 1, got 0

4 tests, 2 failures
```

Notice that the reporting line that shows BATS_RUN_COMMAND for each of the incorrectly-passing tests demonstrates that the `=<exit-status>` is injected as a prefix to the actual command the user wanted to test.

And because (to a reasonable approximation) ~100.00% of machines *lack* a command named `=0`/`=1`/etc, the fact that `run` (by design) masks the exit code of the erroneously-missing `=0`/`=1` command, anyone using this syntax to test either success or failure will not have been testing -- **or even executing** -- the command they thought they were testing, since #479 was merged in October 2021.

```
#!/usr/bin/env bats

# Testing with this file should result in 4x failing tests.
# Instead, it will result in 2x failing and 2x passing tests.

report() {
  echo "# ## Command that 'run' ran was: '$BATS_RUN_COMMAND'" >&3
}

# 2x incorrectly passing tests

@test "This test should fail, but will succeed, #1" {
  run =0 false
  report
}
@test "This test should fail, but will succeed, #2" {
  run =1 true
  report
}

# 2x correctly failing tests, exactly the same as the previous 2 tests, but
# with "=<N>" changed to "-<N>"

@test "This test should fail, and will fail, #1" {
  run -0 false
  report
}
@test "This test should fail, and will fail, #2" {
  run -1 true
  report
}
```

**Expected behavior**

I expected 4 test failures. 

**Environment (please complete the following information):**

- Bats Version 1.6.0 @ `5c7a96727535a91fcef50b22783f0a54e10821eb`
- OS:

```
Distributor ID:	Debian
Description:	Debian GNU/Linux 11 (bullseye)
Release:	11
Codename:	bullseye
```
- Bash version: `GNU bash, version 5.1.4(1)-release (x86_64-pc-linux-gnu)`

**Additional context**

I have opened PR #579 to change the "Writing Tests" documentation to stop pointing users to the old syntax, and to use the current syntax.

**Comment by jpluscplusm** [comment]
On digging further, I believe this is purely a documentation bug, but still one with significant implications for folks who have constructed tests based on those docs! 

I had previously found #367, where `=N` was introduced, but I hadn't spotted #479 which not only added `-N` **but also removed `=N`**.

`=N` is the only convenience method explicitly mentioned in the "Writing Tests" docs, whereas `-N` is mentioned in passing, in examples. Thus, I think my PR #579 actually *is* the fix, here, but I'm unsure how much effort should be put into warning folks who are using `=N` based on the docs that their tests may be invisibly broken.

**Comment by martin-schulze-vireso** [comment]
This is really unfortunate. I obviously missed this Mist crucial instances of the old pattern. 

Anyways, this issue has made me acutely aware of a much bigger oversight: when you run tests that were written with the new run flags in an older bats version, they will simply be executed. This is especially bad with the exit code checks.

**Comment by martin-schulze-vireso** [comment]
Preventing the forward compalitibilty issue (`run` just eating the `-flag`) with older versions is really hard. I can only think of two solutions right now:

1. use another function instead of run, to avoid this; this means we would have to reduce `run` to the old state in newer versions which would be a breaking change and require a major version upgrade
2. introduce a `bats_require_minimum_version` function (#556) that ensures the tests don't run with older versions. However, we'd also need to inform users when they use newer functions without requiring the correct minimum (#547).

**Comment by martin-schulze-vireso** [comment]
Okay, this is my plan now:

1. have new versions of bats generate a warning when `run`'s command is not found (can be suppressed via expecting return code 127)
2. also trigger a warning if a `run` flag is used without requiring its minimum version
3. update old documentation (preferably "inplace" without a bugfix release)

**Comment by martin-schulze-vireso** [comment]
# What happened

1. Due to an incomplete refactoring, the old format for `run`'s convenience exit code check syntax remained in the documentation.
2. Since run interprets everything that does not look like a flag as part of the command, the (wrong) exit code check `=N` will be interpreted as a command. This command likely won't exist on your system, thus triggering an error ("Command not found") and returning exit code 127.
3. Since the exit code check also does not run, due to the wrong format, and a redundant check of `$status` is unlikely, the error can go unnoticed.

# How do I know I am affected? How can this be prevented in the future?

If you want to know if your code is affected, you can run it with Bats version 1.7.0, which contains the warning [BW01](https://bats-core.readthedocs.io/en/latest/warnings/BW01.html). However, note that this can only check actively executed `run`'s. So if your run is `skip`ped or after a failure, it won't be checked.

Furthermore, the documentation for version 1.5.0 and 1.6.0 have been disabled and replaced by documentation that highlights the problem.

# How can this be prevented in the past?

While we can't time travel, your code can still be executed with older versions that don't provide the check mentioned above. However, you can add `bats_require_minimum_version` (also available beginning with version 1.7.0) to your code. On older versions, this command will error since it does not exist. On future versions it will allow feature switches or improved warnings like [BW02](https://bats-core.readthedocs.io/en/latest/warnings/BW02.html)
